src/resource_manager/secret.py

```diff
@@ -25,43 +23,54 @@
 logger = logging.getLogger(__name__)
 
 
+class CertificateDataNotReadyError(CharmStateValidationBaseError):
+    """Exception raised when certificates data is not set on the tls provider."""
+
+
 @dataclasses.dataclass
 class SecretResourceDefinition(ResourceDefinition):
     """A part of charm state with information required to manage secret resource.
 
-    It consistS of 3 components:
-        - SecretResourceInfomation
-        - CharmConfig
-        - TLSInformation
-
     Attrs:
-        external_hostname: The configured gateway hostname.
-        tls_requirer_integration: The integration instance with a TLS provider.
-        tls_certs: A dict of hostname: certificate obtained from the relation.
-        tls_keys: A dict of hostname: private_key stored in juju secrets.
+        hostname: The configured gateway hostname.
         secret_resource_name_prefix: Prefix of the secret resource name.
+        certificate: TLS certificate.
+        private_key: Password-ecrypted private key.
+        password: Private key password.
     """
 
-    external_hostname: str
-    tls_requirer_integration: Relation
-    tls_certs: dict[str, str]
-    tls_keys: dict[str, dict[str, str]]
+    hostname: str
     secret_resource_name_prefix: str
+    certificate: str
+    private_key: str
+    password: str
 
-    def __init__(
-        self,
-        gateway_resource_information: GatewayResourceInformation,
-        charm_config: CharmConfig,
-        tls_information: TLSInformation,
-    ):
-        """Create the state object with state components.
+    @classmethod
+    def from_tls_information(
+        cls, tls_information: TLSInformation, hostname: str
+    ) -> "SecretResourceDefinition":
+        """Get certificate information for a given hostname.
 
         Args:
-            gateway_resource_information: GatewayResourceInformation state component.
-            charm_config: CharmConfig state component.
             tls_information: TLSInformation state component.
+            hostname: The requested hostname.
+
+        Raises:
+            CertificateDataNotReadyError: When the certificate data is not ready.
+
+        Returns:
+            SecretResourceDefinition: Information about the certificate.
         """
-        super().__init__(gateway_resource_information, charm_config, tls_information)
+        if hostname not in tls_information.tls_certs.keys():
```

Comment:
```suggestion
        if hostname not in tls_information.tls_certs:
```

if its a dict , using `in` without `keys()` should suffice